### PR TITLE
add interval to fetch and set remote config

### PR DESCRIPTION
**Pull request by celestial-vault**
### Description

Fetch and set remote config on an interval

### Test Procedure

Change the remote config in the dashboard and confirm that it's applied.

Try edge cases like the request failing or the remote config being disabled.

### Type of Change

-   [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
-   [x] ✨ New feature (non-breaking change which adds functionality)
-   [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] ♻️ Refactor Changes
-   [ ] 💅 Cosmetic Changes
-   [ ] 📚 Documentation update
-   [ ] 🏃 Workflow Changes

### Pre-flight Checklist

-   [x] Changes are limited to a single feature, bugfix or chore (split larger changes into separate PRs)
-   [x] Tests are passing (`npm test`) and code is formatted and linted (`npm run format && npm run lint`)
-   [ ] I have created a changeset using `npm run changeset` (required for user-facing changes)
-   [x] I have reviewed [contributor guidelines](https://github.com/cline/cline/blob/main/CONTRIBUTING.md)

### Additional Notes

----

> [!IMPORTANT]
> Add periodic fetching and application of remote configuration in `Controller` with cache handling and edge case management.
> 
>   - **Behavior**:
>     - Adds periodic fetching of remote config every 30 seconds in `Controller` class in `index.ts`.
>     - Handles edge cases: request failures, no active organization, and disabled remote config.
>   - **Functions**:
>     - `fetchRemoteConfig` in `fetch.ts` updated to apply config to `StateManager` and handle cache operations.
>     - `deleteRemoteConfigFromCache` added to `disk.ts` to remove config from cache if disabled.
>   - **Misc**:
>     - Timer setup and disposal in `Controller` constructor and `dispose()` method in `index.ts`.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=cline%2Fcline&utm_source=github&utm_medium=referral)<sup> for 6b46e1f8a29ac734efff1fc405f55ad141d0f322. You can [customize](https://app.ellipsis.dev/cline/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: src/core/controller/index.ts [hunk 1]
```diff
@@ -87,15 +91,28 @@ export class Controller {
...
+			fetchRemoteConfig(this).catch((error) => {
+				console.error("Failed to fetch remote config:", error)
+			})
+		}, 30000) // 30 seconds
+	}
+
 	constructor(readonly context: vscode.ExtensionContext) {
 		PromptRegistry.getInstance() // Ensure prompts and tools are registered
 		HostProvider.get().logToChannel("ClineProvider instantiated")
 		this.stateManager = StateManager.get()
```
Comment (celestial-vault) [inline src/core/controller/index.ts:115]: Move statemanager callback registrations before these other initializations in case of saving error.

File: src/core/storage/remote-config/fetch.ts [hunk 2]
```diff
@@ -13,12 +15,15 @@ import { applyRemoteConfig } from "./utils"
  * @returns Promise resolving to the RemoteConfig object, or undefined if no active organization exists
  * @throws Error if both API fetch and cache retrieval fail (when an organization exists)
  */
-export async function fetchRemoteConfig(): Promise<RemoteConfig | undefined> {
```
Comment (celestial-vault) [inline src/core/storage/remote-config/fetch.ts:16]: adding controller to be able to call postStateToWebview when config is received.

File: src/core/storage/remote-config/fetch.ts [hunk 3]
```diff
@@ -74,6 +79,12 @@ export async function fetchRemoteConfig(): Promise<RemoteConfig | undefined> {
 
 		// Check if config is enabled
```
Comment (celestial-vault) [inline src/core/storage/remote-config/fetch.ts:80]: Clean up remote config if it goes from enabled --> disabled

File: src/core/storage/remote-config/fetch.ts [hunk 4]
```diff
@@ -13,12 +15,15 @@ import { applyRemoteConfig } from "./utils"
...
  * @throws Error if both API fetch and cache retrieval fail (when an organization exists)
  */
-export async function fetchRemoteConfig(): Promise<RemoteConfig | undefined> {
+export async function fetchRemoteConfig(controller: Controller): Promise<RemoteConfig | undefined> {
 	const authService = AuthService.getInstance()
 
 	// Get the active organization ID
 	const organizationId = authService.getActiveOrganizationId()
+
 	if (!organizationId) {
```
Comment (celestial-vault) [inline src/core/storage/remote-config/fetch.ts:24]: cleanup remote config if user goes from org --> private account

File: src/core/storage/remote-config/fetch.ts [hunk 5]
```diff
@@ -89,23 +100,30 @@ export async function fetchRemoteConfig(): Promise<RemoteConfig | undefined> {
 		// Apply config to StateManager
 		applyRemoteConfig(validatedConfig)
```
Comment (celestial-vault) [inline src/core/storage/remote-config/fetch.ts:102]: notify webview

File: src/core/storage/remote-config/fetch.ts [hunk 6]
```diff
@@ -89,23 +100,30 @@ export async function fetchRemoteConfig(): Promise<RemoteConfig | undefined> {
...
 
+		controller.postStateToWebview()
+
 		return validatedConfig
 	} catch (error) {
 		console.error("Failed to fetch remote config from API:", error)
 
 		// Try to fall back to cached config
 		const cachedConfig = await readRemoteConfigFromCache(organizationId)
 		if (cachedConfig) {
```
Comment (celestial-vault) [inline src/core/storage/remote-config/fetch.ts:111]: add try/catch around validation

File: src/core/storage/disk.ts [hunk 7]
```diff
@@ -315,6 +315,18 @@ export async function writeRemoteConfigToCache(organizationId: string, config: R
 	}
 }
```
Comment (celestial-vault) [inline src/core/storage/disk.ts:317]: cleanup function
